test/support/assert_matches.ex

```diff
@@ -0,0 +1,111 @@
+defmodule Plausible.AssertMatches do
+  @moduledoc false
+
+  @doc """
+  Custom assertion function that allows for flexible matching of expected results.
+
+  This is useful when testing APIs, where some response fields are not deterministic.
+
+  Examples:
+
+      iex> assert_matches %{ key: 1 }, %{ key: expect_any(:integer) }
+
+      iex> assert_matches %{ string: "abc" }, %{ string: "abc" }
+
+      iex> # assert_matches %{ key: 1 }, %{ key: expect_any(:string) }
+      # This would raise ExUnit.AssertionError with a diff.
+
+  """
+  def assert_matches(data, expected) do
+    {equivalent?, left, right} = check_matches(data, expected)
+
+    if not equivalent? do
+      raise ExUnit.AssertionError,
+        left: left,
+        right: right,
+        message: "Values did not match"
+    end
+  end
+
+  defp check_matches(value, expected) do
+    case {value, expected} do
+      # Main case: If expected value is a function, we assume it's a predicate passed from expect_any
+      {value, predicate_fn} when not is_function(value) and is_function(predicate_fn) ->
+        if predicate_fn.(value) do
+          # :TRICKY: To make ex_unit not highlight the predicate_fn, pretend it was the value passed from left
+          {true, value, value}
+        else
+          {false, value, predicate_fn}
+        end
+
+      {value, expected} when is_list(value) and is_list(expected) ->
+        match_lists(value, expected)
+
+      {value, expected} when is_map(value) and is_map(expected) ->
+        match_maps(value, expected)
+
+      {a, b} when is_tuple(a) and is_tuple(b) ->
+        {equivalent?, left, right} = match_lists(Tuple.to_list(a), Tuple.to_list(b))
+        {equivalent?, List.to_tuple(left), List.to_tuple(right)}
+
+      {value, expected} ->
+        {value == expected, value, expected}
+    end
+  end
+
+  defp match_lists([], []), do: {true, [], []}
+  defp match_lists([], right), do: {false, [], right}
+  defp match_lists(left, []), do: {false, left, []}
+
+  defp match_lists([left | left_rest], [right | right_rest]) do
+    {equivalent?, left_acc, right_acc} = match_lists(left_rest, right_rest)
+    {eq, l, r} = check_matches(left, right)
+    {equivalent? and eq, [l | left_acc], [r | right_acc]}
+  end
+
+  defp match_maps(left, expected) do
+    keys =
+      Map.keys(left)
+      |> Enum.concat(Map.keys(expected))
```

Comment:
```suggestion
      |> Enum.concat(Map.keys(expected))
      |> Enum.uniq()
```

to save on redundant checks?